rule/rule.go

```diff
@@ -118,6 +147,34 @@ func LoadWorkspaceData(path, pkg string, data []byte) (*File, error) {
 	return ScanAST(pkg, ast), nil
 }
 
+// LoadMacroData parses a bzl file from a byte slice and scans for the load
+// statements and the rules called from the given Starlark function. If there is
+// no matching function name, then a new function with that name will be created.
+// The function's syntax tree will be returned within File and can be modified by
+// Sync and Save calls.
+func LoadMacroData(path, pkg, defName string, data []byte, isEmpty bool) (*File, error) {
```

Comment:
I don't think the `isEmpty` parameter is needed. If there isn't any function named `defName`, we should create one and add it to `ast.Stmt` next time `Sync()` is called. It's okay if the file is empty or if there's other stuff in there already (`load` statements).

Ideally, `LoadMacroData` should just parse the file, then return `ScanASTBody(pkg, defName, ast)`, similarly to how `LoadData` and `ScanAST` are related.